packages/nx/src/command-line/release/config/deep-merge-json.ts

```diff
@@ -0,0 +1,28 @@
+function isObject(obj: any): obj is Record<string, any> {
+  return obj && typeof obj === 'object' && !Array.isArray(obj);
+}
+
+export function deepMergeJson<T extends Record<string, any>>(
+  target: T,
+  source: T
+): T {
+  try {
+    // Ensure both objects are valid JSON before attempting to merge values
+    JSON.parse(JSON.stringify(source));
```

Comment:
I think this check here would pass for numbers, strings, booleans etc only for below code to fail. Maybe thats fine, just pointing out :)

Reply:
I think this plus the author time checks from TS should be good enough